monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -227,6 +227,17 @@ object Consumer {
   def raiseError[In, R](ex: Throwable): Consumer.Sync[In, R] =
     new RaiseErrorConsumer(ex)
 
+  /** A consumer that will return a collection of all the streamed
+   * values as a Scala `List`.
+   *
+   * Note that this method is functionally the same as the `.toListL` from [[Observable]]
+   *
+   * WARNING: For infinite streams the process will eventually
+   * blow up with an out of memory error.
+   */
+  def collect[A]: Consumer.Sync[A, List[A]] =
+    new FoldLeftConsumer[A, List[A]](() => List.empty[A], (l: List[A], elem: A) => l :+ elem)
```

Comment:
I feel like we could follow `toListL` implementation:
```
foldLeftL(mutable.ListBuffer.empty[A])(_ += _).map(_.toList)
```

I expect `map(_.toList)` to be much cheaper than appending to the end of list each time